formats/json/jvmMain/src/kotlinx/serialization/json/internal/CharArrayPool.kt

```diff
@@ -0,0 +1,24 @@
+package kotlinx.serialization.json.internal
+
+import java.util.concurrent.*
+
+internal object CharArrayPool {
+    private val arrays = ArrayDeque<CharArray>()
+    private var charsTotal = 0
+    private val MAX_CHARS_IN_POOL = runCatching {
+        System.getProperty("kotlinx.serialization.json.pool.size").toIntOrNull()
```

Comment:
This property should be documented somewhere. Maybe not in the .md files, but at least in the JSON documentation.